lib/iris/tests/unit/fileformats/nc_load_rules/actions/test__time_coords.py

```diff
@@ -0,0 +1,459 @@
+            # Translate this into one of timedim_name/perioddim_name
+            general_opts[f"{which}dim_name"] = coord_dim_name
+
+        period_opts = None
+        time_opts = None
+        if which == "time":
+            time_opts = opts
+        else:
+            period_opts = opts
+
+        result = super().run_testcase(
+            time_opts=time_opts, period_opts=period_opts, **general_opts
+        )
+
+        return result
+
+    def check_result(self, cube, coord_is="dim"):
+        """
+        Specialise 'check_result' for single-coord 'time' or 'period' testing.
+        """
+        # Pass generic 'coord_is' option to parent as time/period options.
+        which = self.which
+        assert which in ("time", "period")
+
+        if which == "time":
+            time_is = coord_is
+            period_is = "missing"
+        else:
+            period_is = coord_is
+            time_is = "missing"
+
+        super().check_result(cube, time_is=time_is, period_is=period_is)
+
+    #
+    # Generic single-coordinate testcases.
+    # ( these are repeated for both 'time' and 'time_period' )
+    #
+
+    def test_dimension(self):
+        # Coord is a normal dimension --> dimcoord
+        #
+        # Rules Triggered:
+        #     001 : fc_default
+        #     002 : fc_provides_coordinate_(time[[_period]])
+        #     003 : fc_build_coordinate_(time[[_period]])
+        result = self.run_testcase()
+        self.check_result(result, "dim")
+
+    def test_dimension_in_phenom_coords(self):
+        # Dimension coord also present in phenom:coords.
+        # Strictly wrong but a common error in datafiles : must tolerate.
+        #
+        # Rules Triggered:
+        #     001 : fc_default
+        #     002 : fc_provides_coordinate_(time[[_period]])
+        #     003 : fc_build_coordinate_(time[[_period]])
+        result = self.run_testcase(in_phenomvar_coords=True)
+        self.check_result(result, "dim")
+
+    def test_dim_nonmonotonic(self):
+        # Coord has all-zero values, which prevents it being a dimcoord.
+        # The rule has a special way of treating it as an aux coord
+        # -- even though it doesn't appear in the phenom coords.
+        # ( Done by  the build_coord routine, so not really a rules issue).
+        #
+        # Rules Triggered:
+        #     001 : fc_default
+        #     002 : fc_provides_coordinate_(time[[_period]])
+        #     003 : fc_build_coordinate_(time[[_period]])
+        msg = "Failed to create.* dimension coordinate"
+        result = self.run_testcase(values_all_zero=True, warning=msg)
+        self.check_result(result, "aux")
+
+    def test_dim_fails_typeident(self):
+        # The coord variable is identified as a CFDimensionCoordinate by cf.py,
+        # but having the wrong units causes it to fail the 'is_time' or
+        # 'is_period' test, so the 'provides_coord' rule fails to trigger.
+        # So it is built as a 'miscellaneous' dim-coord.
```

Comment:
This comment should probably be updated also to match the changes in `test_aux_fails_typeident`.